src/lexer.c

```diff
@@ -0,0 +1,102 @@
+#include "minishell.h"
+#include <libft.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <readline/readline.h>
+#include <readline/history.h>
+
+static int	get_sym(char *line, int i, t_tokenized *tokenized)
+{
+	if (line[i] == '>' && line[i + 1] == '>')
+	{
+		tokenized->token = RED_OUT_APPEND;
+		tokenized->element = ">>";
+		return (2);
+	}
+	else if (line[i] == '<' && line[i + 1] == '<')
+	{
+		tokenized->token = RED_IN;
+		tokenized->element = ">>";
+		return (2);
+	}
+	else if (line[i] == '<')
+	{
+		tokenized->token = RED_INPUT;
+		tokenized->element = "<";
+		return (1);
+	}
+	else if (line[i] == '>')
+	{
+		tokenized->token = RED_OUTPUT;
+		tokenized->element = ">";
+		return (1);
+	}
+	else if (line[i] == '|')
+	{
+		tokenized->token = PIPE;
+		tokenized->element = "|";
+		return (1);
+	}
+	return (0);
+}
+
+static int	get_quoted(char *line, int i, t_tokenized *tokenized)
+{
+	int	token;
+
+	token = line[i];
+	if (!ft_strchr(&line[i + 1], token) || !line[i + 1])
+	{
+		printf("ERROR: No closing quote\n");
+		tokenized->token = UNCLOSED;
+		tokenized->element = NULL;
+		return (0);
```

Comment:
Incorrect return value, I think it should move the 'cursur (i)' to the last position so the next time it will return and END token